<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        //error의 전파
         function p(ms){
            return new Promise((resolve, reject)=>{
                setTimeout(()=>{
                   // resolve(ms);
                   reject(new Error('reasonㅅ'))
                }, ms);
            });
        }

       async function asyncP(){
           const ms = await p(1000);
           //error 밑에 진행 안됨(쓰로우 발생)
           //넘기고 싶다면 try catch사용
           return 'Mark :' + ms;
       } 

       (async function main(){
            try{
                const name = await asyncP(1000);
                console.log(name);
            } catch(error){
                console.log(error)
            }
        })();
    </script>
</body>
</html>